plugins/node/instrumentation-react-native-navigation/src/utils/hooks/useTrace.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { MutableRefObject, useEffect, useRef } from 'react';
+import {
+  trace,
+  Tracer,
+  TracerOptions,
+  TracerProvider,
+} from '@opentelemetry/api';
+import { PACKAGE_NAME, PACKAGE_VERSION } from './../../version';
+
+export type TracerRef = MutableRefObject<Tracer | null>;
+
+const useTrace = (
+  provider?: TracerProvider,
+  tracerOptions?: TracerOptions
+): TracerRef => {
+  const tracerRef = useRef<Tracer | null>(null);
+
+  // using the layout effect to make sure the tracer is initialized before the component is rendered
+  useEffect(() => {
+    if (tracerRef.current === null) {
+      if (!provider) {
+        console.info('No TracerProvider found. Using global tracer instead.');
```

Comment:
wondering if we should remove the `console.info` lines in this file for the final version, or maybe adding a configuration to allow users to suppress the logging?

Reply:
I added a `debug` prop into the `config` shape. if this is passed as "true", it will print the console messages. otherwise it won't do anything